libdeno/binding.cc

```diff
@@ -334,6 +342,67 @@ void DenoIsolate::ClearModules() {
   mods_by_name_.clear();
 }
 
+// TODO(kevinkassimo): super duper ugly and cause libdeno to depend on TS...
+void ConsoleLog(v8::Local<v8::Context> context, v8::Local<v8::Value> value) {
+  auto* isolate = context->GetIsolate();
+  DenoIsolate* d = static_cast<DenoIsolate*>(isolate->GetData(0));
+  v8::Isolate::Scope isolate_scope(isolate);
+  v8::HandleScope handle_scope(isolate);
+  v8::Context::Scope context_scope(context);
+
+  auto console = d->console_.Get(isolate);
+  auto logField = console->Get(context, v8_str("log")).ToLocalChecked();
+  if (logField->IsFunction()) {
+    auto logFunc = v8::Local<v8::Function>::Cast(logField);
+    v8::Local<v8::Value> args[1];
+    args[0] = value;
+    logFunc->Call(context->Global(), 1, args);
+  }
+}
+
+void ReplEval(v8::Local<v8::Context> context, const char* js_source) {
+  auto* isolate = context->GetIsolate();
+  DenoIsolate* d = static_cast<DenoIsolate*>(isolate->GetData(0));
+  v8::Isolate::Scope isolate_scope(isolate);
+  v8::HandleScope handle_scope(isolate);
+  v8::Context::Scope context_scope(context);
+
+  d->repl_last_exception_ = "";
+
+  auto source = v8_str(js_source);
+  v8::TryCatch try_catch(isolate);
+
+  auto name = v8_str("<unknown>");
+  v8::ScriptOrigin origin(name);
+  auto script = v8::Script::Compile(context, source, &origin);
+
+  if (script.IsEmpty()) {
+    DCHECK(try_catch.HasCaught());
+    auto exception = try_catch.Exception();
+    d->repl_last_exception_ = EncodeExceptionAsJSON(context, exception, true);
+    return;
+  }
+
+  auto result = script.ToLocalChecked()->Run(context);
+
+  if (result.IsEmpty()) {
+    DCHECK(try_catch.HasCaught());
+    auto exception = try_catch.Exception();
+    // If not native error, print and abort.
+    if (!exception->IsNativeError()) {
+      std::cout << "Thrown: ";
+      ConsoleLog(context, exception);
+      return;
+    }
+    d->repl_last_exception_ = EncodeExceptionAsJSON(context, exception, false);
+    return;
+  }
+
+  // TODO(kevinkassimo): this is nasty,
+  // but we need formatting of value printed...
+  ConsoleLog(context, result.ToLocalChecked());
```

Comment:
Err... I don't like this. 

I liked what you had before `libdeno.evalContext()`  - ideally we could have some thing like this in `js/main.ts`
```ts
window.replMain = (code: string) => {
  let result = libdeno.evalContext(code);
  console.log(result);
};
```

Reply:
This is doable to avoid `ConsoleLog`, but in general the flow would still be controlled on the Rust side. I’m actually more uncertain about the design on the Rust part (since I currently uses some magic numbers to forward REPL specific handling logic to the main thread)